### PR TITLE
Add checks if output files are missing 

### LINKED ISSUE #37: Check if every input file has an output file
In the `run` command, add a check that each input file has a corresponding output file.

### PR DESCRIPTION
**Pull request by MasloMaslane**
Closes #37

### CODE REVIEW

File: src/sinol_make/commands/run/__init__.py [hunk 1]
```diff
@@ -747,6 +748,25 @@ def run(self, args):
             example_tests = [test for test in self.tests if self.get_group(test) == 0]
             if len(example_tests) == len(self.tests):
                 print(util.warning('Running only on example tests.'))
+
+            output_tests = glob.glob(os.path.join(os.getcwd(), "out", "*.out"))
```
Comment (tonowak) [inline src/sinol_make/commands/run/__init__.py:752]: This function is now very long, would it possible to split in into smaller functions?

Comment (MasloMaslane) [reply src/sinol_make/commands/run/__init__.py:752]: Sure

File: src/sinol_make/commands/run/__init__.py [hunk 2]
```diff
@@ -700,6 +705,39 @@ def set_scores(self):
...
+        """
+        output_tests = glob.glob(os.path.join(os.getcwd(), "out", "*.out"))
+        output_tests_ids = [self.extract_test_id(test) for test in output_tests]
+        valid_input_files = []
+        for test in self.tests:
+            if self.extract_test_id(test) in output_tests_ids:
+                valid_input_files.append(test)
+        return valid_input_files
+
+    def validate_tests(self):
```
Comment (tonowak) [inline src/sinol_make/commands/run/__init__.py:720]:
```suggestion
    def validate_existence_of_outputs(self):
```
The current name is misleading (can be related to inwer) and not descriptive of what the function does. I would say that the function can be split further into smaller functions with better names, which would make the code clearer.

Comment (tonowak) [reply src/sinol_make/commands/run/__init__.py:720]: In fact, I'm not sure whether the name I've proposed is even accurate.

Comment (MasloMaslane) [reply src/sinol_make/commands/run/__init__.py:735]: I made some changes, but I didn't come up with a better name for this function

File: src/sinol_make/commands/run/__init__.py [hunk 3]
```diff
@@ -717,24 +717,34 @@ def get_valid_input_files(self):
                 valid_input_files.append(test)
         return valid_input_files
 
+    def validate_existance_of_outputs(self):
```
Comment (tonowak) [inline src/sinol_make/commands/run/__init__.py:720]: https://en.wiktionary.org/wiki/existance

File: src/sinol_make/commands/run/__init__.py [hunk 4]
```diff
@@ -717,24 +717,34 @@ def get_valid_input_files(self):
...
+            missing_tests = list(set(self.tests) - set(valid_input_files))
+            missing_tests.sort()
+
+            print(util.warning('Missing output files for tests: ' + ', '.join(
+                [self.extract_file_name(test) for test in missing_tests])))
+            print(util.warning('Running only on tests with output files.'))
+            self.tests = valid_input_files
+            self.groups = self.get_groups(self.tests)
+
     def validate_tests(self):
```
Comment (tonowak) [inline src/sinol_make/commands/run/__init__.py:735]:
```suggestion
    def check_are_any_tests_to_run(self):
```
Sometimes it's as simple as shortening the description of the function.

**Review by tonowak — APPROVED** [review]
That looks good!

**Pull request merged**